### DISCUSSION TITLE
Angular able pipe - how to use 'or' between subjects?

### DISCUSSION THREAD

**Discussion by jakub-wisniowski-wrx** [stalniy/casl#706] [Q&A]
Hi! I'm using ablePure pipe to check permissions in my template in the following way:

`*ngIf="(appAction.Edit | ablePure: appSubject.CustomerDetails:'addresses':'phone':'mobilePhone' | async)"`

This returns true if user can edit all of the following fields: 'addresses', 'phone' and 'mobilePhone' of CustomerDetails. I also need to check if user can edit at least one of these fields. Splitting them into three calls of ablePure pipe makes the code look messy (in my use case there are 6 fields, not only three). Is there any method to write this check in a concise way?

**Comment by stalniy** [comment]
I'd recommend to rewrite from pipes to `AbilityService` because it allows to use `can` method and logical OR (`||`) without fancy syntax.

Anyway, those pipes are deprecated
